frontend/src/rest/RestClient.js

```diff
@@ -0,0 +1,159 @@
+            headers: {
+                "Authorization": this.authorization,
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify({
+                "title": content.title,
+                "text": content.text,
+                "tags": content.tags
+            })
+        }).then(response => {
+            if(response.status===200)
+                return response.json();
+            return response.text();
+        });
+    }
+
+    handleListAnswersByQuestion(content){
+        return fetch(BASE_URL + "/question/"+content.questionId+"/answers", {
+            method: "GET",
+            headers: {
+                "Authorization": this.authorization
+            }
+        }).then(response => {
+            if(response.status===200)
+                return response.json();
+            return response.text();
+        });
+    }
+
+    handleCreateAnswer(content){
+        return fetch(BASE_URL + "/question/"+content.questionId+"/answers", {
+            method: "POST",
+            headers: {
+                "Authorization": this.authorization,
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify({
+                "text": content.text
+            })
+        }).then(response => {
+            if(response.status===200)
+                return response.json();
+            return response.text();
+        });
+    }
+
+    handleUpdateAnswer(content){
+        return fetch(BASE_URL + "/question/"+content.questionId+"/answers/"+content.answerId, {
+            method:"PUT",
+            headers: {
+                "Authorization": this.authorization,
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify({
+                "text": content.text
+            })
+        }).then(response=>{
+            if(response.status === 200)
+                return response.json();
+            return response.text(); 
+        });
+    }
+
+    handleDeleteAnswer(content){
+        debugger;
+        return fetch(BASE_URL + "/question/"+content.questionId+"/answers/"+content.answerId, {
+            method:"DELETE",
+            headers: {
+                "Authorization": this.authorization,
+            }
+        }).then(response=>{
+            debugger;
+            if(response.status === 200)
+                return "OK";
+            return "NO";
+        });
+    }
+
+    handleCreateQuestionVote(content){
+        return fetch(BASE_URL + "/questions/"+content.questionId+"/vote?isUpVote="+content.type, {
+            method: "POST",
+            headers: {
+                "Authorization": this.authorization
+            }
+        }).then(response => {
+            if(response.status===200)
+                return; //mesaj de ok sau not ok
+            return response.text();
+        });
+    }
+}
+
+const restClient = new RestClient();
+restClient.authenticate("ana","000");
```

Comment:
I would prefer if you also handle the login (and not always use a hard-coded user).